std/assembly/bindings/dom.ts

```diff
@@ -0,0 +1,162 @@
+  export function random(): f64;
+  @external("env", "Math.round")
+  export function round(x: f64): f64;
+  @external("env", "Math.sign")
+  export function sign(x: f64): f64;
+  @external("env", "Math.sin")
+  export function sin(x: f64): f64;
+  @external("env", "Math.sinh")
+  export function sinh(x: f64): f64;
+  @external("env", "Math.sqrt")
+  export function sqrt(x: f64): f64;
+  @external("env", "Math.tan")
+  export function tan(x: f64): f64;
+  @external("env", "Math.tanh")
+  export function tanh(x: f64): f64;
+  @external("env", "Math.trunc")
+  export function trunc(x: f64): f64;
+}
+
+export declare namespace Reflect {
+  @external("env", "Reflect.get")
+  export function get(target: externref, propertyKey: string): externref;
+  @external("env", "Reflect.getWithReceiver")
+  @external.js("return Reflect.get(target, propertyKey, receiver);")
+  export function getWithReceiver(target: externref, propertyKey: string, receiver: externref): externref;
+  @external("env", "Reflect.has")
+  export function has(target: externref, propertyKey: string): bool;
+  @external("env", "Reflect.set")
+  export function set(target: externref, propertyKey: string, value: externref): externref;
+  @external("env", "Reflect.setWithReceiver")
+  @external.js("return Reflect.set(target, propertyKey, value, receiver);")
+  export function setWithReceiver(target: externref, propertyKey: string, value: externref , receiver: externref): externref;
+  @external("env", "Reflect.apply")
+  export function apply(target: externref, thisArgument: externref, argumentsList: externref): externref;
+}
+
+export declare namespace String {
+  @external("env", "String.fromCodePoint")
+  export function fromCodePoint(codepoint: i32): externref;
+  @external("env", "String.fromCodePoints")
+  @external.js("return String.fromCodePoint(...codepoints);")
+  export function fromCodePoints(codepoints: i32[]): externref;
+}
+
+export declare namespace Date {
+  @external("env", "Date.now")
+  export function now(): f64;
+}
+
+export declare namespace console {
+  @external("env", "console.assert")
+  export function assert(condition: bool, message: string): void;
+  @external("env", "console.log")
+  export function log(text: string): void;
+  @external("env", "console.debug")
+  export function debug(text: string): void;
+  @external("env", "console.info")
+  export function info(text: string): void;
+  @external("env", "console.warn")
+  export function warn(text: string): void;
+  @external("env", "console.error")
+  export function error(text: string): void;
+  @external("env", "console.time")
+  export function time(label: string): void;
+  @external("env", "console.timeLog")
+  export function timeLog(label: string): void;
+  @external("env", "console.timeEnd")
+  export function timeEnd(label: string): void;
+}
+
+export declare namespace document {
+  @external("env", "document.getElementById")
+  export function getElementById(id: string): externref;
```

Comment:
Is the assumption that a library like `asdom` would import these namespaces, then export the ones with the actual types? f.e. `function getElementById(id: string): Element | null`, without `externref`, using wrappers around `externref`?

Reply:
This is merely what's possible today, since we cannot yet "describe" DOM capabilities purely in AS code, which is still TODO. See also the previous comment on how this could look in the future :)